Sources/Purchasing/TransactionsManager.swift

```diff
@@ -15,33 +15,14 @@ import StoreKit
 
 class TransactionsManager {
 
-    private let storeKit2Setting: StoreKit2Setting
     private let receiptParser: ReceiptParser
 
-    init(storeKit2Setting: StoreKit2Setting,
-         receiptParser: ReceiptParser) {
-        self.storeKit2Setting = storeKit2Setting
+    init(receiptParser: ReceiptParser) {
         self.receiptParser = receiptParser
     }
 
-    func customerHasTransactions(receiptData: Data, completion: @escaping (Bool) -> Void) {
-        // Note: this uses SK2 (unless it's explicitly disabled) because its implementation is more accurate.
-        if #available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *), self.storeKit2Setting != .disabled {
-            Async.call(with: completion) {
-                return await self.sk2CheckCustomerHasTransactions()
-            }
-        } else {
-            completion(self.sk1CheckCustomerHasTransactions(receiptData: receiptData))
-        }
-    }
-
-    func sk1CheckCustomerHasTransactions(receiptData: Data) -> Bool {
-        receiptParser.receiptHasTransactions(receiptData: receiptData)
-    }
-
-    @available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
-    func sk2CheckCustomerHasTransactions() async -> Bool {
-        await StoreKit.Transaction.all.contains { _ in true }
+    func customerHasTransactions(receiptData: Data) -> Bool {
+        return self.receiptParser.receiptHasTransactions(receiptData: receiptData)
```

Comment:
let's add a note here explaining that we're always using SK1, even though SK2 might potentially be more accurate, because we need to reflect whatever's on the receipt if we're posting SK1 receipts

Reply:
Done.